### PR TITLE
made the use if aws credentials conditional so can run in IAM context

### PR DESCRIPTION
**Pull request by tbenade**
a PR to address this question. https://github.com/apiaryio/s3-streaming-upload/issues/49

I have never written a line of coffeescript apologies. I have tested the code having edited the javascript output on EC2.

Feedback and heckling most welcome.

### CODE REVIEW

File: src/uploader.coffee [hunk 1]
```diff
@@ -7,11 +7,15 @@ class Uploader extends EventEmitter
   # Constructor
   constructor: ({accessKey, secretKey, sessionToken, region, stream, objectName, objectParams, bucket, partSize, maxBufferSize, waitForPartAttempts, waitTime, debug}, @cb) ->
     super()
-    aws.config.update
-      accessKeyId:     accessKey
-      secretAccessKey: secretKey
-      sessionToken:    sessionToken
-      region:          region if region
+
+    if accessKey?
```
Comment (abtris) [inline src/uploader.coffee]: you don't need use `accessKey?` just `accessKey` but on EC2 instances are set in environment variables?

Comment (tbenade) [inline src/uploader.coffee]: Im sorry I am not sure I understand the question but I will have a go. 
 A quote from the documentation.

> If you are running your application on Amazon EC2, you can leverage EC2's IAM roles functionality in order to have credentials automatically provided to the instance.

> If you have configured your instance to use IAM roles, the SDK will automatically select these credentials for use in your application, and you do not need to manually provide credentials in any other format.

http://docs.aws.amazon.com/AWSJavaScriptSDK/guide/node-configuring.html#Credentials_from_IAM_Roles_for_EC2_Instances

My change is simply testing for the existence of the supplied accessKey. If the consumer did not explicitly supply an accessKey it is fair to say they are not attempting to explicitly set the AWS credentials. Therefore don't set any of the values and let the SDK take care of authentication automatically for you. Doest that make more sense? The I did the same for region.

My javascript code looks like this.

```
if (accessKey){
  aws.config.update({
    accessKeyId: accessKey,
    secretAccessKey: secretKey,
    sessionToken: sessionToken,
  });
}
if (region){
  aws.config.update({
    region: region ? region : void 0
  });
}
```

**Comment by abtris** [comment]
We need some test example for this. Now in 0.2.3 you can pass AWS.S3() object for endpoint and others parameters. But I'm not sure about how this can be useful.

**Comment by abtris** [comment]
Tests failing.

**Pull request closed without merging**